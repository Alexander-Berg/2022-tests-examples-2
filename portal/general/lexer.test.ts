import {lex} from './lexer';

describe('lexer', () => {
    it('should lex', () => {
        const code = `<html>
        <!-- [% ignore %]-->
            [% expression %]
        </html>`;
        const tokens = [...lex(code, 'test.html')];
        expect(tokens.length).toBe(1);
        expect(tokens).toMatchSnapshot();
    });

    it('should lex', () => {
        const code = `<html>
            <[% expression %]>
        </html>`;
        const tokens = [...lex(code, 'test.html')];
        expect(tokens.length).toBe(1);
        expect(tokens).toMatchSnapshot();
    });
});
