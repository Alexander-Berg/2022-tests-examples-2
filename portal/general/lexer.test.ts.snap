// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lexer should lex 1`] = `
Array [
  Object {
    "loc": Object {
      "end": Object {
        "column": 25,
        "line": 3,
      },
      "source": "test.html",
      "start": Object {
        "column": 14,
        "line": 3,
      },
    },
    "range": Array [
      50,
      61,
    ],
    "type": 1,
    "value": " expression ",
  },
]
`;

exports[`lexer should lex 2`] = `
Array [
  Object {
    "loc": Object {
      "end": Object {
        "column": 26,
        "line": 2,
      },
      "source": "test.html",
      "start": Object {
        "column": 15,
        "line": 2,
      },
    },
    "range": Array [
      22,
      33,
    ],
    "type": 1,
    "value": " expression ",
  },
]
`;
